### PR TITLE
Add new CVar `sv_usercmd_custom_random_seed`

### PR DESCRIPTION
**Pull request by wopox1337**
| CVar | Def. value | Description | 
|:--|:--:|:--|
| sv_usercmd_custom_random_seed | 0 | When enabled server will populate an additional random seed independent of the client. | 

Ported from [CS:GO leaked source code](https://github.com/perilouswithadollarsign/cstrike15_src/blob/f82112a2388b841d72cb62ca48ab1846dfcc11c8/game/server/player.cpp#L3719)

Thanks to `danil.nesvetaev` for the reminder of this functionality in CS:GO. 👍

### CODE REVIEW

**Comment by hackovh** [comment]
man good nice bro
add hold aim please 

`
	host_client->packet_loss = packet_loss;
	if (!g_psv.paused && (g_psvs.maxclients > 1 || !key_dest) && !(sv_player->v.flags & FL_FROZEN))
	{
		sv_player->v.v_angle[0] = cmds[0].viewangles[0];
		sv_player->v.v_angle[1] = cmds[0].viewangles[1];
		sv_player->v.v_angle[2] = cmds[0].viewangles[2];

#ifdef REHLDS_FIXES
		if (sv_usercmd_holdaim.value && numcmds > 1 && !sv_player->v.deadflag && sv_player->v.health > 0.f)
		{
			usercmd_t *hold_aim_cmd = nullptr;

			for (int i = 0; i < numcmds; i++)
			{
				if (cmds[i].buttons & IN_ATTACK || cmds[i].buttons & IN_ATTACK2)
				{
					hold_aim_cmd = &cmds[i];
					break;
				}
			}

			if (hold_aim_cmd != nullptr)
			{
				Con_DPrintf("Client %s:%s hold aim (x %.3f y %.3f z %.3f) / (x %.3f y %.3f z %.3f)\n", host_client->name, NET_AdrToString(host_client->netchan.remote_address), hold_aim_cmd->viewangles[0], hold_aim_cmd->viewangles[1], hold_aim_cmd->viewangles[2], cmds[0].viewangles[0], cmds[0].viewangles[1], cmds[0].viewangles[2]);
				
				cmds[0].viewangles[0] = hold_aim_cmd->viewangles[0];
				cmds[0].viewangles[1] = hold_aim_cmd->viewangles[1];
				cmds[0].viewangles[2] = hold_aim_cmd->viewangles[2];
			}
		}
#endif // REHLDS_FIXES
	}
	else
	{
		for (int i = 0; i < numcmds; i++)
		{`

**Comment by metita** [comment]
Is there any reason for this change to be implemented? AFAIK having this cvar on 1 will just make the visual bullet landing worse, so...?

**Comment by WPMGPRoSToTeMa** [comment]
@metita no-recoil will still work.

**Comment by metita** [comment]
> @metita no-recoil will still work.

Any pro / cons on this then?

**Comment by WPMGPRoSToTeMa** [comment]
> Any pro / cons on this then?

@metita pros: no-spread will not work. Cons: client-side and server-side gunshot positions are different. E.g. you hit a player on the client-side, but a wall on the server-side (so no decals on the client-side) or vice versa. Or just different positions on a wall.

**Comment by afwn90cj93201nixr2e1re** [comment]
.
